pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +91,611 @@ sub _check_for_libkeyutils {
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_B_rootkit {
+    my ($self) = @_;
+    my $has_cdorked_b = 0;
+    my @files = ( '/usr/sbin/arpd ', '/usr/sbin/tunelp ', '/usr/bin/s2p ' );
+    my $cdorked_files;
+
+    for my $file (@files) {
+        if ( -e $file ) {
+            $has_cdorked_b = 1;
+            $cdorked_files .= "[$file] ";
+        }
+    }
+
+    if ( $has_cdorked_b == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: Evidence of the CDORKED B rootkit was found." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_libkeyutils_filenames {
+    my ($self) = @_;
+    my $bad_libs;
+    my @dirs  = qw( /lib /lib64 );
+    my @files = qw(
+      libkeyutils.so.1.9
+      libkeyutils-1.2.so.0
+      libkeyutils-1.2.so.2
+      libkeyutils.so.1.3.0
+      libkeyutils.so.1.3.2
+      libns2.so
+      libns5.so
+      libpw3.so
+      tls/libkeyutils.so.1
+      tls/libkeyutils.so.1.5
+    );
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        for my $file (@files) {
+            if ( -f "${dir}/${file}" and !-z "${dir}/${file}" ) {
+                $bad_libs .= "\t${dir}/${file}\n";
+            }
+        }
+    }
+
+    if ($bad_libs) {
+        $self->add_bad_advice(
+            'text'       => [ "Libkey rootkit check: The following suspicious system libraries " . $bad_libs . " were found which could indicate a root level compromise." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub build_libkeyutils_file_list {
+    my ($self) = @_;
+    my @dirs = qw( /lib /lib64 );
+    my @libkeyutils_files;
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        opendir( my $dir_fh, $dir );
+        while ( my $file = readdir($dir_fh) ) {
+            if ( $file =~ /^libkeyutils\.so\.(?:[\.\d]+)?$/ ) {
+                if ( $dir eq '/lib' ) {
+                    push @libkeyutils_files, "/lib/$file\n";
+                }
+                elsif ( $dir eq '/lib64' ) {
+                    push @libkeyutils_files, "/lib64/$file\n";
+                }
+            }
+        }
+        closedir $dir_fh;
+    }
+
+    chomp @libkeyutils_files;
```

Comment:
So, we explicitly add a newline to the end of each string we push onto the array, then we chomp them off.  We might as well not add them to begin with (lines 258, 261).